tests/test_main.py

```diff
@@ -81,6 +81,17 @@ def test_path_writable_io_error(mock_os_access):
     with pytest.raises(IOError):
         validate.__main__.path_writable(test_path)
 
+@mock.patch('validate.validators.bam.pysam.quickcheck')
+@mock.patch('validate.validators.bam.pysam.view')
+def test_file_bam_empty(mock_pysam_quickcheck, mock_pysam_view):
+    '''Tests Value error for valid BAM header with no reads'''
+    mock_pysam_quickcheck.return_value = ''
+    mock_pysam_view.return_value = '0\n'
+    test_path = Path('empty/valid/bam')
```

Comment:
You might consider mocking `Path` as well, so any operations performed on it do not actually touch a filesystem.

Reply:
That makes sense! It's used in older unit tests as well so I can clean up the whole file in a separate dedicated PR.